### ISSUE TITLE
checkpoint.run() results in “no attribute” error

### ISSUE THREAD

**Issue by tnorlund-dsc** [fivetran/great_expectations#3437]
**Describe the bug**
I'm able to create a validator using the expectation suite I created. I'm able to run the `validate()` method fine. I'd now like to create a checkpoint and store the results. When I attempt to run the `run()` method, I get a "no attribute" error.

**To Reproduce**
I'm running this in EMR. I'll paste the cells I'm using to explain the steps I've made.

# Install Dependencies
```
sc.install_pypi_package("cython")
sc.install_pypi_package("pandas==0.25.1")
sc.install_pypi_package("matplotlib", "https://pypi.org/simple")
sc.install_pypi_package("boto3")
sc.install_pypi_package("great_expectations")
```
```
import sys
from datetime import datetime
from pyspark.sql.functions import *
# from ruamel import yaml
import yaml

import great_expectations as ge
from great_expectations.core.batch import BatchRequest, RuntimeBatchRequest
from great_expectations.data_context import BaseDataContext
from great_expectations.data_context.types.base import (
    DataContextConfig,
    InMemoryStoreBackendDefaults,
)
from great_expectations.data_context.types.base import DataContextConfig, DatasourceConfig, S3StoreBackendDefaults
from great_expectations.data_context import BaseDataContext
from great_expectations.core import RunIdentifier
from great_expectations.core.batch import BatchRequest, RuntimeBatchRequest
from great_expectations.checkpoint import SimpleCheckpoint, Checkpoint
from great_expectations.exceptions import DataContextError
```
# Initialize Context
```
project_config = DataContextConfig(
    config_version=1,
    plugins_directory=None,
    config_variables_file_path=None,
    datasources={
        "my_spark_datasource": {
            "data_asset_type": {
                "class_name": "SparkDFDataset",
                "module_name": "great_expectations.dataset",
            },
            "class_name": "SparkDFDatasource",
            "module_name": "great_expectations.datasource",
            "batch_kwargs_generators": {},
        }
    },
    stores={
        "expectations_S3_store": {
            "class_name": "ExpectationsStore",
            "store_backend": {
                "class_name": "TupleS3StoreBackend",
                "bucket": "my_s3_bucket",
                "prefix": "my_expectations_store_prefix",
            },
        },
        "validations_S3_store": {
            "class_name": "ValidationsStore",
            "store_backend": {
                "class_name": "TupleS3StoreBackend",
                "bucket": "my_s3_bucket",
                "prefix": "my_validations_store_prefix",
            },
        },
        "checkpoints_S3_store": {
            "class_name": "CheckpointStore",
            "store_backend": {
                "class_name": "TupleS3StoreBackend",
                "bucket": "my_s3_bucket",
                "prefix": "my_checkpoint_store_prefix",
            },
        },
        "evaluation_parameter_store": {
            "class_name": "EvaluationParameterStore"
        },
    },
    expectations_store_name="expectations_S3_store",
    validations_store_name="validations_S3_store",
    checkpoint_store_name="validations_S3_store",
    evaluation_parameter_store_name="evaluation_parameter_store",
    data_docs_sites={
        "s3_site": {
            "class_name": "SiteBuilder",
            "store_backend": {
                "class_name": "TupleS3StoreBackend",
                "bucket": "my_s3_bucket",
                "prefix": "my_optional_data_docs_prefix",
            },
            "site_index_builder": {
                "class_name": "DefaultSiteIndexBuilder",
                "show_cta_footer": True,
            },
        }
    },
    validation_operators={
        "action_list_operator": {
            "class_name":
                "ActionListValidationOperator",
            "action_list": [
                {
                    "name": "store_validation_result",
                    "action": {
                        "class_name": "StoreValidationResultAction"
                    },
                },
                {
                    "name": "store_evaluation_params",
                    "action": {
                        "class_name": "StoreEvaluationParametersAction"
                    },
                },
                {
                    "name": "update_data_docs",
                    "action": {
                        "class_name": "UpdateDataDocsAction"
                    },
                },
            ],
        }
    },
    anonymous_usage_statistics={"enabled": True})
context = BaseDataContext(project_config=project_config)
```
# Read DataFrame from S3
```
status = "credit_balance_changed"
df = spark.read.parquet(f"s3://my_s3_bucket/directory/{status}/")
```
# Create Checkpoint
```
status = "credit_balance_changed"
datasource_config = {
    "name": "myDataSource",
    "class_name": "Datasource",
    "execution_engine": {
        "class_name": "SparkDFExecutionEngine"
    },
    "data_connectors": {
        "default_runtime_data_connector_name": {
            "class_name": "RuntimeDataConnector",
            "batch_identifiers": [
                "batch_id"
            ],
        }
    },
}
checkpoint_name = f"{status}-checkpoint"
expectation_suite_name = status
action_list = [
    {
        'name': 'store_validation_result', 
        'action': {
            'class_name': 'StoreValidationResultAction'
        }
    }, 
    {
        'name': 'store_evaluation_params', 
        'action': {
            'class_name': 'StoreEvaluationParametersAction'
        }
    }, 
    {
        'name': 'update_data_docs', 
        'action': {
            'class_name': 'UpdateDataDocsAction'
        }
    }
]
context.add_datasource(**datasource_config)
batch_request = RuntimeBatchRequest(
    datasource_name = "myDataSource",
    data_connector_name = "default_runtime_data_connector_name",
    data_asset_name = "myDataAsset",  # This can be anything that identifies this data_asset for you
    batch_identifiers = {"batch_id": "example_batch_id"},
    runtime_parameters = {"batch_data": df},  # Your dataframe goes here
)
checkpoint_config = {
    "config_version": 1.0,
    "class_name": "Checkpoint",
    "run_name_template": f"%Y-%M-{checkpoint_name}",
    "validations": [
        {
            "batch_request": batch_request,
            "expectation_suite_name": expectation_suite_name,
            "action_list": action_list,
            "evaluation_parameters": None,
        }
    ],
}
validator = context.get_validator(
    batch_request = batch_request,
    expectation_suite_name = status
)
validation_result = validator.validate()
checkpoint = Checkpoint(
    name = checkpoint_name, 
    data_context = context, 
    **checkpoint_config
)
checkpoint_result = checkpoint.run()
```
> An error was encountered:
'RuntimeBatchRequest' object has no attribute 'items'
Traceback (most recent call last):
  File "/tmp/1632339776101-0/lib/python3.7/site-packages/great_expectations/checkpoint/checkpoint.py", line 279, in run
    validation_dict=validation_dict,
  File "/tmp/1632339776101-0/lib/python3.7/site-packages/great_expectations/checkpoint/util.py", line 248, in get_substituted_validation_dict
    validation_batch_request=validation_dict.get("batch_request"),
  File "/tmp/1632339776101-0/lib/python3.7/site-packages/great_expectations/checkpoint/util.py", line 229, in get_runtime_batch_request
    for key, val in runtime_batch_request_dict.items():
AttributeError: 'RuntimeBatchRequest' object has no attribute 'items'

**Expected behavior**
A checkpoint is created and the data docs are updated with the checkpoint that is run.

**Environment (please complete the following information):**
 - Operating System: AWS EMR
 - Great Expectations Version: 0.13.34

**Additional context**

**Comment by NathanFarmer** [comment]
Hi @tnorlund-dsc! It looks like you have identified a bug. We have placed high priority on this and PR #3438 was opened to address it. Also, I believe you were correct in opening Issue #3404 to address the documentation. I am planning to re-open that issue.

**Comment by NathanFarmer** [comment]
Hi @tnorlund-dsc! Thank you again for identifying this. Bug fix PR #3438 has been merged into `develop` and will be part of the next release.
